app/templates/view_supplier_services.html

```diff
@@ -25,39 +25,49 @@
   {% with heading = "Services" %}
     {% include "toolkit/page-heading.html" %}
   {% endwith %}
-  {% call(item) summary.list_table(
-    services,
-    caption="Services",
-    empty_message="This supplier has no services on the Digital Marketplace",
-    field_headings=[
-      'Name',
-      'Service ID',
-      'Framework',
-      'Lot',
-      'Status',
-      summary.hidden_field_heading("Action")
-    ],
-    field_headings_visible=True
-  ) %}
-    {% call summary.row() %}
-      {{ summary.service_link(item.serviceName, '/' + item.frameworkFramework + '/services/' + item.id|string) }}
-      {{ summary.text(item.id) }}
-      {{ summary.text(item.frameworkName) }}
-      {{ summary.text(item.lotName) }}
-      {% call summary.field() %}
-        <span class="service-status-{{ item.status }}">
-          {% if item.status == "published" %}
-            Public
-          {% elif item.status == "enabled" %}
-            Private
-          {% elif item.status == "disabled" %}
-            Removed
-          {% else %}
-            {{ item.status|title }}
-          {% endif %}
-        </span>
+  {% if not frameworks_services %}
+    {% call(item) summary.list_table(
+      empty_message="This supplier has no services on the Digital Marketplace."
+    )
+    %}{% endcall %}
+  {% endif %}
+  {% for framework in frameworks | sort(attribute='id', reverse=True) %}
+    {% if framework['slug'] in frameworks_services %}
+
+      {{ summary.heading(framework['name'], id="{}_services".format(framework['slug'])) }}
+      {% call(item) summary.list_table(
+        frameworks_services[framework['slug']],
+        caption="Services",
+        empty_message="This supplier has no services on the {} framework.".format(framework['name']),
```

Comment:
This is just for backup/completeness and should never actually happen, right?

Reply:
Yup